### PR TITLE
ToolbarButton: Deprecate `isDisabled` prop and merge with `disabled`

### LINKED ISSUE #63070: ToolbarButton: Accepts both `disabled` and `isDisabled` props with different behavior
ToolbarButton is intended to be disabled with the designated `isDisabled` prop, judging from the code and TypeScript types. 

However, it also accepts the standard `disabled` prop because it allows standard `<button>` props via `restProps`. To complicate things further, the `disabled` prop in these `restProps` are first passed onto `Ariakit.ToolbarItem` before eventually being passed down to `Button`, which opts out the toolbar item from the roving tabindex set regardless of whether the underlying `Button` allows focusable disabled.

So to allow focusable disabled, you'll need to pass a combination of either `disabled` & `accessibleWhenDisabled` (Ariakit API which is undocumented in our docs), or `isDisabled` & `__experimentalIsFocusable`.

I'm not immediately sure how we'd want to address this, but things should be less confusing. We might want to tweak the logic so that passing `disabled` will result in the same behavior as passing `isDisabled`.

---

Related: #57701

### PR DESCRIPTION
**Pull request by mirka**
Follow-up to https://github.com/WordPress/gutenberg/pull/62306#discussion_r1663277829
Fixes #63070

## What?

Deprecates the `isDisabled` prop on `ToolbarButton` and merges it into `disabled`.

## Why?

ToolbarButton accepts both `isDisabled` and `disabled` props to disable the button, and can result in slightly different behavior between the two, especially when wanting to achieve a focusable disabled button (see details in #63070).

Current API usage in our codebase shows that most people reach for the `disabled` prop. I could also find no meaningful significance for the `isDisabled` prop to be there ([original PR where it was introduced](https://github.com/WordPress/gutenberg/pull/20008#discussion_r421248963)).

## How?

[Passive deprecate](https://github.com/WordPress/gutenberg/issues/61099) `isDisabled` and alias it to `disabled`.

## Testing Instructions

Play around with the [Storybook code](https://github.com/WordPress/gutenberg/blob/566b82a13633cb72006d1900931ca46a8e0d7937/packages/components/src/toolbar/stories/index.story.tsx#L118-L121) for the `ToolbarButton`s in the `Toolbar` story. Passing `isDisabled` should result in the same behavior as passing `disabled`, especially in combination with accessible disabled props like `__experimentalIsFocusable` or `accessibleWhenDisabled`.

### CODE REVIEW

File: packages/components/src/toolbar/toolbar-button/index.tsx [hunk 1]
```diff
@@ -19,51 +19,60 @@ import ToolbarButtonContainer from './toolbar-button-container';
...
 						event: ReactMouseEvent<
 							HTMLButtonElement & HTMLAnchorElement,
 							MouseEvent
 						>
 					) => {
 						event.stopPropagation();
-						// TODO: Possible bug; maybe use onClick instead of props.onClick.
-						if ( props.onClick ) {
-							props.onClick( event );
+						// TODO: Possible bug; maybe use onClick instead of restProps.onClick.
```
Comment (ciampo) [inline packages/components/src/toolbar/toolbar-button/index.tsx:63]: Unrelated, but what is this comment about? 🤔

**Review by ciampo — APPROVED** [review]
LGTM.

One thing that I noticed while testing is that, if I pass `disabled` and `__experimentalIsFocusable` to `ToolbarButton`, the disabled button is still skipped when using arrow keys, but is not skipped when using the Tab key.

File: packages/components/src/toolbar/toolbar-button/index.tsx (hunk 1 above)
Comment (mirka) [reply packages/components/src/toolbar/toolbar-button/index.tsx:63]: The `onClick` function written here will be overridden by the consumer's `onClick` function passed through with the `restProps` 🙃

**Comment by mirka** [comment]
> One thing that I noticed while testing is that, if I pass `disabled` and `__experimentalIsFocusable` to `ToolbarButton`, the disabled button is still skipped when using arrow keys, but is not skipped when using the Tab key.

Good catch 😳 I confirmed this also happens in trunk, but I think it will be normalized by #63102.

**Pull request merged**